hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/erasurecode/StripedBlockReconstructor.java

```diff
@@ -126,12 +128,26 @@ private void reconstructTargets(int toReconstructLen) throws IOException {
     int[] erasedIndices = stripedWriter.getRealTargetIndices();
     ByteBuffer[] outputs = stripedWriter.getRealTargetBuffers(toReconstructLen);
 
+    if (isValidationEnabled()) {
+      markBuffers(inputs);
+      decode(inputs, erasedIndices, outputs);
+      resetBuffers(inputs);
+
+      DataNodeFaultInjector.get().badDecoding(outputs);
+      getValidator().validate(inputs, erasedIndices, outputs);
```

Comment:
How about handle the InvalidDecodingException separately? I think this situation need more attention.

Reply:
@runitao
Thanks for your comment!
How about adding a metric for the exception like `EcInvalidReconstructionTasks`? (I saw your deleted comment.)
As for logging, I think it's better to output more messages through the entire EC reconstruction process, and so I'd like to handle it in another issue.
Are you suggesting anything else?

Reply:
+1，I have no other suggestion.

Reply:
@runitao Fixed at 5eeeaf5.